File: qt/gen_qatomic.go
package qt

import (
	"unsafe"
)

type QAtomicInt struct {
	h          uintptr
	isSubclass bool
}

// NewQAtomicInt constructs a new QAtomicInt object.
func NewQAtomicInt() *QAtomicInt {

	ret := newQAtomicInt(QAtomicInt_new())
	ret.isSubclass = true
	return ret
}

// NewQAtomicInt2 constructs a new QAtomicInt object.
func NewQAtomicInt2(param1 *QAtomicInt) *QAtomicInt {

	ret := newQAtomicInt(QAtomicInt_new2(param1.cPointer()))
	ret.isSubclass = true
	return ret
}

// NewQAtomicInt3 constructs a new QAtomicInt object.
func NewQAtomicInt3(value int) *QAtomicInt {

	ret := newQAtomicInt(QAtomicInt_new3((int)(value)))
	ret.isSubclass = true
	return ret
}

func (this *QAtomicInt) OperatorAssign(param1 *QAtomicInt) {
	QAtomicInt_OperatorAssign(this.h, param1.cPointer())
}

File: qt/gen_qatomic.cpp
// +build ignore

#include <QAtomicInt>
#include <qatomic.h>
#include "gen_qatomic.h"

QAtomicInt* QAtomicInt_new() {
	return new QAtomicInt();
}

QAtomicInt* QAtomicInt_new2(QAtomicInt* param1) {
	return new QAtomicInt(*param1);
}

QAtomicInt* QAtomicInt_new3(int value) {
	return new QAtomicInt(static_cast<int>(value));
}

void QAtomicInt_OperatorAssign(QAtomicInt* self, QAtomicInt* param1) {
	self->operator=(*param1);
}

void QAtomicInt_Delete(QAtomicInt* self, bool isSubclass) {
	if (isSubclass) {
		delete dynamic_cast<QAtomicInt*>( self );
	} else {
		delete self;
	}
}

